controllers/storageclaim_controller.go

```diff
@@ -398,15 +411,28 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 				// storageclaim is clusterscoped resources using its
 				// hash as the clusterID
 				data["clusterID"] = r.storageClaimHash
-				if resource.Name == "cephfs" {
-					volumeSnapshotClass = r.getCephFSVolumeSnapshotClass(data)
-				} else if resource.Name == "ceph-rbd" {
-					volumeSnapshotClass = r.getCephRBDVolumeSnapshotClass(data)
-				}
+				volumeSnapshotClass = r.getCephDriverVolumeSnapshotClass(resource.Name, data)
+				utils.AddLabel(volumeSnapshotClass, storageIDRamenLabel, r.OperatorNamespace)
 				utils.AddAnnotation(volumeSnapshotClass, storageClaimAnnotation, r.storageClaim.Name)
-				if err := r.createOrReplaceVolumeSnapshotClass(volumeSnapshotClass); err != nil {
+				if err := r.createOrReplaceClass(volumeSnapshotClass); err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeSnapshotClass: %s", err)
 				}
+			case "VolumeGroupSnapshotClass":
+				// check for CRD availability
+				if r.availableCRDs["VolumeGroupSnapshotClass"] {
+					var volumeGroupSnapshotClass *groupsnapapi.VolumeGroupSnapshotClass
+					data["csi.storage.k8s.io/group-snapshotter-secret-namespace"] = r.OperatorNamespace
+					// generate a new clusterID for cephfs subvolumegroup, as
+					// storageclaim is clusterscoped resources using its
+					// hash as the clusterID
+					data["clusterID"] = r.storageClaimHash
+					volumeGroupSnapshotClass = r.getCephDriverVolumeGroupSnapshotClass(resource.Name, data)
+					utils.AddLabel(volumeGroupSnapshotClass, storageIDRamenLabel, storageID)
+					utils.AddAnnotation(volumeGroupSnapshotClass, storageClaimAnnotation, r.storageClaim.Name)
+					if err := r.createOrReplaceClass(volumeGroupSnapshotClass); err != nil {
```

Comment:
when the storageclaim gets deleted we need to ensure that no volumegroupsnapshotcontent exists before deleting the storageclaim